src/Symfony/Component/Console/Application.php

```diff
@@ -941,6 +959,17 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             // ignore invalid options/arguments for now, to allow the event listeners to customize the InputDefinition
         }
 
+        if ($this->signalRegistry) {
+            $event = new ConsoleSignalEvent($command, $input, $output);
+            $onStopHandler = function () use ($event) {
```

Comment:
1. according to your code, you get the signumber here, let's use it (see my previous comment)
2. onStopHandler is not representative. `onSignalHandler`  seems better